shared/js/background/message-handlers.js

```diff
@@ -145,12 +145,27 @@ export function getTopBlockedByPages (options) {
 }
 
 // Click to load interactions
-export async function initClickToLoad (config, sender) {
+export async function initClickToLoad (contentScopeConfig, sender) {
     await settings.ready()
     const tab = tabManager.get({ tabId: sender.tab.id })
 
     // Remove first-party entries.
     await startup.ready()
+
+    /**
+     * The Click to Load configuration is in the process of being migrated, but it is currently split
+     * between tdsStorage.ClickToLoadConfig (aka social_ctp_configuration.json), the block list (aka tds.json),
+     * and content-scope-scripts/features/click-to-play.js.
+     * Here we take care to remove any entries from the content-scope-scripts configuration (aka contentScopeConfig)
+     * that aren't also found in tdsStorage.ClickToLoadConfig - otherwise the feature can end up in an inconsistent state.
+     */
+    const config = {}
+    for (const entity in contentScopeConfig) {
+        if (tdsStorage.ClickToLoadConfig[entity]) {
+            config[entity] = contentScopeConfig[entity]
+        }
+    }
+
```

Comment:
Remove this in favor of the tds config to be rolled out soon.

Reply:
I don't fully understand this, would be nice to add context to the above comment @ladamski so it's explained properly. But I know you two synced on it, so will click Approve again. The changes otherwise LGTM.

Reply:
Good point @kzar! I just left some notes during the call so I didn't forget what needed to be updated, but this one would benefit from more context indeed. I've added an update in the original thread https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/1554#discussion_r1038052473